api/genericresource/validate.go

```diff
@@ -0,0 +1,53 @@
+package genericresource
+
+import (
+	"github.com/docker/swarmkit/api"
+)
+
+// ValidateTask validates that the task only uses integers
+// for Third Party Resources
+func ValidateTask(resources *api.Resources) error {
+	for k, v := range resources.Generic {
+		switch v.Resource.(type) {
+		case *api.GenericResource_Integer:
+			continue
+		case *api.GenericResource_Set:
+			return NewError("Invalid Argument for resource %s", k)
+		}
+	}
+
+	return nil
+}
+
+func getGenericResourceInteger(res *api.GenericResource) (*api.GenericResource_Integer, error) {
+	if _, ok := res.Resource.(*api.GenericResource_Integer); !ok {
+		return nil, NewError("Task should only hold Integer type Third Party Resources")
```

Comment:
Error messages should start with lowercase

Change "Third Party Resources" to "generic resources"